src/api/services/PendingValidationInfoDispatcher.ts

```diff
@@ -0,0 +1,166 @@
+            .setStyle(ButtonStyle.Danger)
+            .setCustomId("delete");
+
+        const buttonRow =
+            new ActionRowBuilder<MessageActionRowComponentBuilder>().addComponents(
+                verifyButton, deleteButton
+            );
+        const message = await channelTOPostTo.send({
+            embeds: [infoEmbed],
+            components: [buttonRow]
+        });
+        const collector = message.createMessageComponentCollector();
+        collector.on("collect", async (collectInteraction: ButtonInteraction) => {
+            try {
+                await collectInteraction.deferUpdate();
+                const buttonId = collectInteraction.customId;
+
+                let ok = false;
+                let messageStr = "";
+                if (buttonId === "delete") {
+                    ok = await this.deleteSubmission(payload.id);
+                    messageStr = "deleted";
+                } else if (buttonId === "verify") {
+                    ok = await this.verifySubmission(payload.id);
+                    messageStr = "verified";
+                }
+
+                if (ok) {
+                    const followUp = await collectInteraction.followUp({
+                        content: `Submission has been ${messageStr} successfully`
+                    });
+                    await message.delete();
+                    setTimeout(() => {
+                        followUp.delete().catch();
+                    }, 3000);
+                } else {
+                    const followup = await collectInteraction.followUp({
+                        content: "Unable to delete"
+                    });
+                    setTimeout(() => {
+                        followup.delete().catch();
+                    }, 3000);
+                }
+            } catch (e) {
+                console.error(e);
+            }
+        });
+        return message;
+    }
+
+    private async deleteSubmission(id: number): Promise<boolean> {
+        const response = await fetch(`${this.webappUrl}/rest/submission/deleteEntries`, {
+            method: "DELETE",
+            headers: {
+                "Authorization": this.getAuthHeader(),
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify([id])
+        });
+        if (!response.ok) {
+            console.error(response.statusText);
+        }
+        return response.ok;
+    }
+
+    private async verifySubmission(id: number): Promise<boolean> {
+        const response = await fetch(`${this.webappUrl}/rest/submission/verifyEntries`, {
+            method: "POST",
+            headers: {
+                "Authorization": this.getAuthHeader(),
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify([id])
+        });
+        if (!response.ok) {
+            console.error(response.statusText);
+        }
+        return response.ok;
+    }
+
+    private getAuthHeader(): string {
```

Comment:
if you want, this could be expanded to return the whole headers object since that seems to be the same for both of these calls.